### DISCUSSION TITLE
Is it possible to use dot notation to reference nested variables in scope?

### DISCUSSION THREAD

**Discussion by simoami** [josdejong/mathjs#3131] [Q&A] [answered]
I would like to find out if there is a way to namespace variables passed to the scope so that variables can be accessed using (.) notation. For example:
```typescript
const scope = new Map();
scope.set('Vitals', { Weight: math.unit(70, "kg"), Height: math.unit(1.5, "m")});
const bmiOutput = math.evaluate('Vitals.Weight / Vitals.Height ^ 2', scope);
```
This would be ideal for IDE auto completion in domains where the number of variables is massive (i.e. over 1000 variables).
In clinical studies, the typical structure to reference a variable within a study is `StudyVersionID.FormID.ItemGroupID.ItemID`

**Comment by josdejong** [comment, marked as answer]
Dot notation is supported, your code actually just works (except for the JavaScript typos in there)

**Reply by simoami** [reply]
Ah ok, glad to hear, thank you. 
ps: corrected the typo for others finding this thread.
